task_test.go

```diff
@@ -37,6 +37,18 @@ type SyncBuffer struct {
 	mu  sync.Mutex
 }
 
+func (sb *SyncBuffer) String() string {
+	sb.mu.Lock()
+	defer sb.mu.Unlock()
+	return sb.buf.String()
+}
+
+func (sb *SyncBuffer) Reset() {
+	sb.mu.Lock()
+	defer sb.mu.Unlock()
+	sb.buf.Reset()
+}
+
```

Comment:
I don't think these are necessary. You can just call `buff.buf.Reset()` like you have done elsewhere. The only operation that needs a mutex is `Write()`.